openedx/core/djangoapps/content_staging/api.py

```diff
@@ -63,10 +178,39 @@ def get_user_clipboard_json(user_id: int, request: HttpRequest | None = None):
     except _UserClipboard.DoesNotExist:
         # This user does not have any content on their clipboard.
         return {"content": None, "source_usage_key": "", "source_context_title": "", "source_edit_url": ""}
-    serializer = _UserClipboardSerializer(clipboard, context={'request': request})
+    serializer = _UserClipboardSerializer(
+        _user_clipboard_model_to_data(clipboard),
+        context={'request': request},
+    )
     return serializer.data
 
 
+def _user_clipboard_model_to_data(clipboard: _UserClipboard) -> UserClipboardData:
+    """
+    Convert a UserClipboard model instance to an immutable data object.
+    """
+    content = clipboard.content
+    source_context_key = clipboard.source_usage_key.context_key
+    if source_context_key.is_course and (course_overview := get_course_overview_or_none(source_context_key)):
+        source_context_title = course_overview.display_name_with_default
+    else:
+        source_context_title = str(source_context_key)  # Fall back to stringified context key as a title
```

Comment:
This logic was extracted out of `models.py:UserClipboard`.

I was on the fence between moving it here (so that UserClipboardData.source_context_title is always available) vs moving it to the serializer (so that the CourseOverview is only loaded upon serialization). Reviewers, let me know if you'd like it on the serializer instead, and I could move it there.

Reply:
Why can't we just leave `get_source_context_title` on the django model, so we can use it in the django admin? and then in this function here we just need to say `source_context_title=clipboard.get_source_context_title()`

Reply:
Oh. Duh. Yeah, I'll do that 😅

Reply:
Done